src/MongoDB.Driver/PipelineStageDefinitionBuilder.cs

```diff
@@ -1286,6 +1286,47 @@ public static class PipelineStageDefinitionBuilder
             return SortByCount(new ExpressionAggregateExpressionDefinition<TInput, TValue>(value, translationOptions));
         }
 
+        /// <summary>
+        /// Creates a $unionWith stage.
+        /// </summary>
+        /// <typeparam name="TInput">The type of the input documents.</typeparam>
+        /// <typeparam name="TForeignDocument">The type of the foreign collection documents.</typeparam>
+        /// <typeparam name="TOutput">The type of the output documents.</typeparam>
+        /// <param name="foreignCollection">The foreign collection.</param>
+        /// <param name="unionWithPipeline">The unionWith pipeline.</param>
+        /// <param name="options">The options.</param>
+        /// <returns>The stage.</returns>
+        public static PipelineStageDefinition<TInput, TOutput> UnionWith<TInput, TForeignDocument, TOutput>(
+            IMongoCollection<TForeignDocument> foreignCollection,
```

Comment:
Maybe there is a better name here. Thoughts?

Reply:
Yes, let's use:

    withCollection
    withPipeline